File: python/leetcode/leet_0387.py
# First Unique Character in a String
# https://leetcode.com/problems/first-unique-character-in-a-string/description/?envType=daily-question&envId=2024-02-05

def first_unique_char(s: str) -> int:
    char_count = {}

    for char in s:
        char_count[char] = char_count.get(char, 0) + 1

    for i, char in enumerate(s):
        if char_count[char] == 1:
            return i

    return -1

File: rust/leet_code/src/leet_0387.rs
// First Unique Character in a String
// https://leetcode.com/problems/first-unique-character-in-a-string/?envType=daily-question&envId=2024-02-05

use std::collections::HashMap;

fn first_uniq_char(s: String) -> i32 {
    let mut char_count: HashMap<char, i32> = HashMap::new();

    for ch in s.chars() {
        *char_count.entry(ch).or_insert(0) += 1;
    }

    for (i, ch) in s.chars().enumerate() {
        if char_count[&ch] == 1 {
            return i as i32;
        }
    }
    -1
}
